ape_etherscan/dependency.py

```diff
@@ -0,0 +1,64 @@
+from ape.api.projects import DependencyAPI
+from ape.exceptions import ProjectError
+from ape.types import AddressType
+from ethpm_types import PackageManifest
+from hexbytes import HexBytes
+from pydantic import AnyUrl, HttpUrl, field_validator
+
+from .explorer import Etherscan
+
+
+class EtherscanDependency(DependencyAPI):
+    etherscan: str
+    ecosystem: str = "ethereum"
+    network: str = "mainnet"
+
+    @field_validator("etherscan", mode="before")
+    @classmethod
+    def handle_int(cls, value):
+        return value if isinstance(value, str) else HexBytes(value).hex()
+
+    @property
+    def version_id(self) -> str:
+        return "etherscan"  # Only 1 version
```

Comment:
do we want to use `network` and `ecosystem` somehow here?

Reply:
that is a good idea!! that would actually work too.